src/main/java/apoc/load/LoadXls.java

```diff
@@ -172,28 +178,57 @@ private Object convertArray(Object value) {
-                case FLOAT: return Util.toDouble(value);
-                case BOOLEAN: return Util.toBoolean(value);
-                case NULL: return null;
-                case LIST: return Arrays.stream(arrayPattern.split(value.toString())).map(this::convertType).collect(Collectors.toList());
-                default: return value;
+            switch (type) { // Destination Type
+                case STRING:
+                    if (value instanceof TemporalAccessor && !dateFormat.isEmpty()) {
+                        return formatDate((TemporalAccessor) value, dateFormat);
+                    } else {
+                        return value.toString();
+                    }
+                case INTEGER:
+                    return Util.toLong(value);
+                case FLOAT:
+                    return Util.toDouble(value);
+                case BOOLEAN:
+                    return Util.toBoolean(value);
+                case NULL:
+                    return null;
+                case LIST:
+                    return Arrays.stream(arrayPattern.split(value.toString())).map(this::convertType).collect(Collectors.toList());
+                case DATE:
+                    return parseDate(value.toString(), LocalDate.class, dateParse);
+                case DATE_TIME:
+                    return parseDate(value.toString(), ZonedDateTime.class, dateParse);
+                case LOCAL_DATE_TIME:
+                    return parseDate(value.toString(), LocalDateTime.class, dateParse);
+                case LOCAL_TIME:
+                    return parseDate(value.toString(), LocalTime.class, dateParse);
+                case TIME:
+                    return parseDate(value.toString(), OffsetTime.class, dateParse);
+                case DURATION:
+                    return formatDuration(value.toString());
+                default:
+                    return value;
             }
         }
     }
 
+    private static String[] convertFormat(Object value) {
+        if (value == null) return null;
+        List<String> strings = (List<String>) value;
+        return strings.toArray(new String[strings.size()]);
+    }
+
     private String[] getHeader(boolean hasHeader, Row header, Selection selection, List<String> ignore, Map<String, Mapping> mapping) throws IOException {
         if (!hasHeader) return null;
 
         String[] result = new String[selection.right - selection.left];
         for (int i = selection.left; i < selection.right; i++) {
             Cell cell = header.getCell(i);
             if (cell == null) throw new IllegalStateException("Header at position "+i+" doesn't have a value");
-            String value = cell.getStringCellValue();
-            result[i- selection.left] = ignore.contains(value) || mapping.getOrDefault(value, Mapping.EMPTY).ignore ? null : value;
+//            String value = cell.getStringCellValue();
+            Object value = getValue(cell, cell.getCellTypeEnum());
```

Comment:
Questo è sbagliato qui stiamo prendendo gli header, che devono sempre essere stringa.
Perché facciamo questo?
Rimetterlo come prima

Reply:
l'ho modificato perché se nel caso in cui nell'header ho una data (=DATA(2018;01;01)) con il metodo vecchio va in eccezione